src/filesystem/FileSystem.c

```diff
@@ -88,11 +86,12 @@ void filesystemAddNewFileSystemEntry(filesystemConfiguration_t *filesystemConfig
     filesystemConfig->fileSystem[filesystemConfig->numberOfEntries].entry.size = size;
     filesystemConfig->fileSystem[filesystemConfig->numberOfEntries].entry.isConfig = isConfig;
 
-    // Set number of used sectors for this file
+    uint32_t numberOfUsedSectors = (size_t)ceilf(
```

Comment:
not 100% sure rn, but these lines and the lines above seem to all handle a very specific aspect of the configuration. It might make sense to simplify the use here by wrapping them in functions. E.g. having

```c
static inline void set_size(FsConfig *config, size_t size) {
  config->fileSystem[config->numberOfEntries].entry.size = size;
}
```

and afterwards just writing
```c
set_size(filesystemConfig, size);
```

makes the code a lot easier to digest.